The system cannot find the file specified

Hi

Could someone please help me with the following error:

```
C:\git\databot2\databot\src\za\co\resbank>lein inline-deps
project prefix:  mrandersonb765aca7
retrieve dependencies and munge clojure source files
in RESOLVED-TREE mode, working on a resolved dependency tree
 [clj-http "3.10.0"]
   [commons-io "2.6" :exclusions [[org.clojure/clojure]]]
   [org.apache.httpcomponents/httpmime "4.5.8" :exclusions [[org.clojure/clojure]]]
   [potemkin "0.4.5" :exclusions [[org.clojure/clojure]]]
     [clj-tuple "0.2.2"]
     [riddley "0.1.12"]
   [org.apache.httpcomponents/httpclient "4.5.8" :exclusions [[org.clojure/clojure]]]
     [commons-logging "1.2"]
   [org.apache.httpcomponents/httpcore "4.4.11" :exclusions [[org.clojure/clojure]]]
   [slingshot "0.12.2" :exclusions [[org.clojure/clojure]]]
   [org.apache.httpcomponents/httpclient-cache "4.5.8" :exclusions [[org.clojure/clojure]]]
   [org.apache.httpcomponents/httpasyncclient "4.1.4" :exclusions [[org.clojure/clojure]]]
     [org.apache.httpcomponents/httpcore-nio "4.4.10"]
   [commons-codec "1.12" :exclusions [[org.clojure/clojure]]]
unzipping [ clj-http  [ v3v10v0 ]]
unzipping [ httpmime  [ v4v5v8 ]]
unzipping [ commons-io  [ v2v6 ]]
unzipping [ potemkin  [ v0v4v5 ]]
unzipping [ clj-tuple  [ v0v2v2 ]]
unzipping [ httpclient-cache  [ v4v5v8 ]]
unzipping [ httpasyncclient  [ v4v1v4 ]]
unzipping [ httpclient  [ v4v5v8 ]]
unzipping [ commons-logging  [ v1v2 ]]
unzipping [ commons-codec  [ v1v12 ]]
unzipping [ httpcore-nio  [ v4v4v10 ]]
unzipping [ httpcore  [ v4v4v11 ]]
unzipping [ slingshot  [ v0v12v2 ]]
unzipping [ riddley  [ v0v1v12 ]]
  munge source files of riddley artifact on branch [] exposed false.
java.io.FileNotFoundException: target\default\srcdeps\riddley\compiler.clj (The system cannot find the file specified)
```

My `project.clj` looks as follows:

```
(defproject za.co.resbank.databot "0.1.0-SNAPSHOT"
  :description "FIXME: write description"
  :url "http://example.com/FIXME"
  :license {:name "EPL-2.0 OR GPL-2.0-or-later WITH Classpath-exception-2.0"
            :url "https://www.eclipse.org/legal/epl-2.0/"}
  :dependencies [[org.clojure/clojure "1.10.0"]
                 [org.clojure/data.csv "0.1.4"]
                 [cheshire "5.8.1"]
                 ^:inline-dep [clj-http "3.10.0"]
                 [seesaw "1.5.0"]
                 [za.co.resbank/haverapi-java "2.1.0.0"]
                 [com.bloomberglp/blpapi "3.12.1-1"]
                 [clojure.java-time "0.3.2"]]
  :main ^:skip-aot za.co.resbank.databot
  :target-path "target/%s"
  :profiles {:uberjar {:aot :all}}
  :plugins [[thomasa/mranderson "0.5.1"]])
```

I made a few stupid mistakes, having resolved them things look like this, I am unable to run the repl on package the JAR without an error from lein:

```
Microsoft Windows [Version 10.0.17134.523]
(c) 2018 Microsoft Corporation. All rights reserved.

C:\git\databot2\databot>lein do clean

C:\git\databot2\databot>lein inline-deps
project prefix:  mranderson9a2f6379
retrieve dependencies and munge clojure source files
in RESOLVED-TREE mode, working on a resolved dependency tree
 [cheshire "5.8.1"]
   [com.fasterxml.jackson.core/jackson-core "2.9.6"]
   [com.fasterxml.jackson.dataformat/jackson-dataformat-cbor "2.9.6"]
   [com.fasterxml.jackson.dataformat/jackson-dataformat-smile "2.9.6"]
   [tigris "0.1.1"]
 [clj-http "3.10.0"]
   [commons-io "2.6" :exclusions [[org.clojure/clojure]]]
   [org.apache.httpcomponents/httpmime "4.5.8" :exclusions [[org.clojure/clojure]]]
   [potemkin "0.4.5" :exclusions [[org.clojure/clojure]]]
     [riddley "0.1.12"]
   [org.apache.httpcomponents/httpclient "4.5.8" :exclusions [[org.clojure/clojure]]]
     [commons-logging "1.2"]
   [org.apache.httpcomponents/httpcore "4.4.11" :exclusions [[org.clojure/clojure]]]
   [slingshot "0.12.2" :exclusions [[org.clojure/clojure]]]
   [org.apache.httpcomponents/httpclient-cache "4.5.8" :exclusions [[org.clojure/clojure]]]
   [org.apache.httpcomponents/httpasyncclient "4.1.4" :exclusions [[org.clojure/clojure]]]
     [org.apache.httpcomponents/httpcore-nio "4.4.10"]
   [commons-codec "1.12" :exclusions [[org.clojure/clojure]]]
 [clojure.java-time "0.3.2"]
   [clj-tuple "0.2.2"]
 [com.bloomberglp/blpapi "3.12.1-1"]
 [org.clojure/data.csv "0.1.4"]
 [seesaw "1.5.0"]
   [com.fifesoft/rsyntaxtextarea "2.5.6"]
   [com.jgoodies/forms "1.2.1"]
   [com.miglayout/miglayout "3.7.4"]
   [j18n "1.0.2"]
   [org.swinglabs.swingx/swingx-core "1.6.3"]
     [org.swinglabs.swingx/swingx-action "1.6.3"]
     [org.swinglabs.swingx/swingx-autocomplete "1.6.3"]
       [org.swinglabs.swingx/swingx-common "1.6.3"]
     [org.swinglabs.swingx/swingx-painters "1.6.3"]
     [org.swinglabs.swingx/swingx-plaf "1.6.3"]
 [za.co.resbank/haverapi-java "2.1.0.0"]
   [net.java.dev.jna/jna "5.3.1"]
unzipping [ seesaw  [ v1v5v0 ]]
unzipping [ cheshire  [ v5v8v1 ]]
unzipping [ jackson-dataformat-smile  [ v2v9v6 ]]
unzipping [ clj-http  [ v3v10v0 ]]
unzipping [ httpmime  [ v4v5v8 ]]
unzipping [ commons-io  [ v2v6 ]]
unzipping [ potemkin  [ v0v4v5 ]]
unzipping [ clojurejava-time  [ v0v3v2 ]]
unzipping [ clj-tuple  [ v0v2v2 ]]
unzipping [ swingx-core  [ v1v6v3 ]]
unzipping [ swingx-autocomplete  [ v1v6v3 ]]
unzipping [ tigris  [ v0v1v1 ]]
unzipping [ swingx-painters  [ v1v6v3 ]]
unzipping [ swingx-plaf  [ v1v6v3 ]]
unzipping [ swingx-common  [ v1v6v3 ]]
unzipping [ forms  [ v1v2v1 ]]
unzipping [ haverapi-java  [ v2v1v0v0 ]]
unzipping [ jna  [ v5v3v1 ]]
unzipping [ httpclient-cache  [ v4v5v8 ]]
unzipping [ httpasyncclient  [ v4v1v4 ]]
unzipping [ httpclient  [ v4v5v8 ]]
unzipping [ datacsv  [ v0v1v4 ]]
unzipping [ commons-logging  [ v1v2 ]]
unzipping [ commons-codec  [ v1v12 ]]
unzipping [ rsyntaxtextarea  [ v2v5v6 ]]
unzipping [ miglayout  [ v3v7v4 ]]
unzipping [ jn  [ v1v0v2 ]]
unzipping [ httpcore-nio  [ v4v4v10 ]]
unzipping [ httpcore  [ v4v4v11 ]]
unzipping [ slingshot  [ v0v12v2 ]]
unzipping [ jackson-dataformat-cbor  [ v2v9v6 ]]
unzipping [ jackson-core  [ v2v9v6 ]]
unzipping [ blpapi  [ v3v12v1-1 ]]
unzipping [ swingx-action  [ v1v6v3 ]]
unzipping [ riddley  [ v0v1v12 ]]
  munge source files of riddley artifact on branch [] exposed false.
  munge source files of swingx-action artifact on branch [] exposed false.
  munge source files of blpapi artifact on branch [] exposed false.
  munge source files of jackson-core artifact on branch [] exposed false.
  munge source files of jackson-dataformat-cbor artifact on branch [] exposed false.
  munge source files of slingshot artifact on branch [] exposed false.
  munge source files of httpcore artifact on branch [] exposed false.
  munge source files of httpcore-nio artifact on branch [] exposed false.
  munge source files of jn artifact on branch [] exposed false.
  munge source files of miglayout artifact on branch [] exposed false.
  munge source files of rsyntaxtextarea artifact on branch [] exposed false.
  munge source files of commons-codec artifact on branch [] exposed false.
  munge source files of commons-logging artifact on branch [] exposed false.
  munge source files of datacsv artifact on branch [] exposed false.
  munge source files of httpclient artifact on branch [] exposed false.
  munge source files of httpasyncclient artifact on branch [] exposed false.
  munge source files of httpclient-cache artifact on branch [] exposed false.
  munge source files of jna artifact on branch [] exposed false.
  munge source files of haverapi-java artifact on branch [] exposed false.
  munge source files of forms artifact on branch [] exposed false.
  munge source files of swingx-common artifact on branch [] exposed false.
  munge source files of swingx-plaf artifact on branch [] exposed false.
  munge source files of swingx-painters artifact on branch [] exposed false.
  munge source files of tigris artifact on branch [] exposed false.
  munge source files of swingx-autocomplete artifact on branch [] exposed false.
  munge source files of swingx-core artifact on branch [] exposed false.
  munge source files of clj-tuple artifact on branch [] exposed false.
    new ns: mranderson9a2f6379.clj-tuple.v0v2v2.clj-tuple
  munge source files of clojurejava-time artifact on branch [] exposed false.
    new ns: mranderson9a2f6379.clojurejava-time.v0v3v2.java-time
  munge source files of potemkin artifact on branch [] exposed false.
    new ns: mranderson9a2f6379.potemkin.v0v4v5.potemkin
  munge source files of commons-io artifact on branch [] exposed false.
  munge source files of httpmime artifact on branch [] exposed false.
  munge source files of clj-http artifact on branch [] exposed false.
  munge source files of jackson-dataformat-smile artifact on branch [] exposed false.
  munge source files of cheshire artifact on branch [] exposed false.
  munge source files of seesaw artifact on branch [] exposed false.

C:\git\databot2\databot>lein with-profile +plugin.mranderson/config uberjar
Compiling za.co.resbank.bloomberg
Syntax error compiling at (bloomberg.clj:1:1).
Exception in thread "main" Syntax error compiling at (bloomberg.clj:1:1).
        at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3707)
        ...
Caused by: java.io.FileNotFoundException: Could not locate java_time__init.class, java_time.clj or java_time.cljc on classpath. Please check that namespa
ces with dashes use underscores in the Clojure file name.
        ...
Compilation failed: Subprocess failed
Uberjar aborting because jar failed: Compilation failed: Subprocess failed
Error encountered performing task 'uberjar' with profile(s): 'base,system,user,provided,dev,config'
Uberjar aborting because jar failed: Compilation failed: Subprocess failed
```

Okay. On a fresh project `lein new app mranderson-test` with the only added  (inlined) dependency being `clj-http`, `lein with-profile +plugin.mranderson/config uberjar` runs successfully. 

As soon as I require `[clj-http.client :as client]` in the `ns` form, it no longer packages the JAR:

```
C:\git\mranderson-test>lein with-profile +plugin.mranderson/config uberjar
Compiling mranderson-test.core
Syntax error compiling at (core.clj:1:1).
Exception in thread "main" Syntax error compiling at (core.clj:1:1).
        at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3707)
        at clojure.lang.Compiler.compile1(Compiler.java:7730)
        at clojure.lang.Compiler.compile1(Compiler.java:7720)
        at clojure.lang.Compiler.compile(Compiler.java:7797)
        at clojure.lang.RT.compile(RT.java:415)
        at clojure.lang.RT.load(RT.java:461)
        at clojure.lang.RT.load(RT.java:428)
        at clojure.core$load$fn__6824.invoke(core.clj:6126)
        at clojure.core$load.invokeStatic(core.clj:6125)
        at clojure.core$load.doInvoke(core.clj:6109)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invokeStatic(core.clj:5908)
        at clojure.core$compile$fn__6829.invoke(core.clj:6136)
        at clojure.core$compile.invokeStatic(core.clj:6136)
        at clojure.core$compile.invoke(core.clj:6128)
        at user$eval155$fn__164.invoke(form-init1340189419723536912.clj:1)
        at user$eval155.invokeStatic(form-init1340189419723536912.clj:1)
        at user$eval155.invoke(form-init1340189419723536912.clj:1)
        at clojure.lang.Compiler.eval(Compiler.java:7176)
        at clojure.lang.Compiler.eval(Compiler.java:7166)
        at clojure.lang.Compiler.eval(Compiler.java:7166)
        at clojure.lang.Compiler.load(Compiler.java:7635)
        at clojure.lang.Compiler.loadFile(Compiler.java:7573)
        at clojure.main$load_script.invokeStatic(main.clj:452)
        at clojure.main$init_opt.invokeStatic(main.clj:454)
        at clojure.main$init_opt.invoke(main.clj:454)
        at clojure.main$initialize.invokeStatic(main.clj:485)
        at clojure.main$null_opt.invokeStatic(main.clj:519)
        at clojure.main$null_opt.invoke(main.clj:516)
        at clojure.main$main.invokeStatic(main.clj:598)
        at clojure.main$main.doInvoke(main.clj:561)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.lang.Var.applyTo(Var.java:705)
        at clojure.main.main(main.java:37)
Caused by: java.io.FileNotFoundException: Could not locate clj_http/client__init.class, clj_http/client.clj or clj_http/client.cljc on classpath. Please
check that namespaces with dashes use underscores in the Clojure file name.
        at clojure.lang.RT.load(RT.java:466)
        at clojure.lang.RT.load(RT.java:428)
        at clojure.core$load$fn__6824.invoke(core.clj:6126)
        at clojure.core$load.invokeStatic(core.clj:6125)
        at clojure.core$load.doInvoke(core.clj:6109)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invokeStatic(core.clj:5908)
        at clojure.core$load_one.invoke(core.clj:5903)
        at clojure.core$load_lib$fn__6765.invoke(core.clj:5948)
        at clojure.core$load_lib.invokeStatic(core.clj:5947)
        at clojure.core$load_lib.doInvoke(core.clj:5928)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.core$load_libs.invokeStatic(core.clj:5985)
        at clojure.core$load_libs.doInvoke(core.clj:5969)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.core$require.invokeStatic(core.clj:6007)
        at clojure.core$require.doInvoke(core.clj:6007)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at mranderson_test.core$loading__6706__auto____171.invoke(core.clj:1)
        at clojure.lang.AFn.applyToHelper(AFn.java:152)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3702)
        ... 33 more
Compilation failed: Subprocess failed
Uberjar aborting because jar failed: Compilation failed: Subprocess failed
Error encountered performing task 'uberjar' with profile(s): 'base,system,user,provided,dev,config'
Uberjar aborting because jar failed: Compilation failed: Subprocess failed
```

Like so?

```
C:\git\mranderson-test>lein do clean, inline-deps, with-profile +plugin.mranderson/config uberjar
project prefix:  mrandersondc3099b1
retrieve dependencies and munge clojure source files
in RESOLVED-TREE mode, working on a resolved dependency tree
 [clj-http "3.10.0"]
   [commons-io "2.6" :exclusions [[org.clojure/clojure]]]
   [org.apache.httpcomponents/httpmime "4.5.8" :exclusions [[org.clojure/clojure]]]
   [potemkin "0.4.5" :exclusions [[org.clojure/clojure]]]
     [clj-tuple "0.2.2"]
     [riddley "0.1.12"]
   [org.apache.httpcomponents/httpclient "4.5.8" :exclusions [[org.clojure/clojure]]]
     [commons-logging "1.2"]
   [org.apache.httpcomponents/httpcore "4.4.11" :exclusions [[org.clojure/clojure]]]
   [slingshot "0.12.2" :exclusions [[org.clojure/clojure]]]
   [org.apache.httpcomponents/httpclient-cache "4.5.8" :exclusions [[org.clojure/clojure]]]
   [org.apache.httpcomponents/httpasyncclient "4.1.4" :exclusions [[org.clojure/clojure]]]
     [org.apache.httpcomponents/httpcore-nio "4.4.10"]
   [commons-codec "1.12" :exclusions [[org.clojure/clojure]]]
unzipping [ clj-http  [ v3v10v0 ]]
unzipping [ httpmime  [ v4v5v8 ]]
unzipping [ commons-io  [ v2v6 ]]
unzipping [ potemkin  [ v0v4v5 ]]
unzipping [ clj-tuple  [ v0v2v2 ]]
unzipping [ httpclient-cache  [ v4v5v8 ]]
unzipping [ httpasyncclient  [ v4v1v4 ]]
unzipping [ httpclient  [ v4v5v8 ]]
unzipping [ commons-logging  [ v1v2 ]]
unzipping [ commons-codec  [ v1v12 ]]
unzipping [ httpcore-nio  [ v4v4v10 ]]
unzipping [ httpcore  [ v4v4v11 ]]
unzipping [ slingshot  [ v0v12v2 ]]
unzipping [ riddley  [ v0v1v12 ]]
  munge source files of riddley artifact on branch [] exposed false.
  munge source files of slingshot artifact on branch [] exposed false.
  munge source files of httpcore artifact on branch [] exposed false.
  munge source files of httpcore-nio artifact on branch [] exposed false.
  munge source files of commons-codec artifact on branch [] exposed false.
  munge source files of commons-logging artifact on branch [] exposed false.
  munge source files of httpclient artifact on branch [] exposed false.
  munge source files of httpasyncclient artifact on branch [] exposed false.
  munge source files of httpclient-cache artifact on branch [] exposed false.
  munge source files of clj-tuple artifact on branch [] exposed false.
    new ns: mrandersondc3099b1.clj-tuple.v0v2v2.clj-tuple
  munge source files of potemkin artifact on branch [] exposed false.
    new ns: mrandersondc3099b1.potemkin.v0v4v5.potemkin
  munge source files of commons-io artifact on branch [] exposed false.
  munge source files of httpmime artifact on branch [] exposed false.
  munge source files of clj-http artifact on branch [] exposed false.
Compiling mranderson-test.core
Syntax error compiling at (core.clj:1:1).
Exception in thread "main" Syntax error compiling at (core.clj:1:1).
...
Caused by: java.io.FileNotFoundException: Could not locate clj_http/client__init.class, clj_http/client.clj or clj_http/client.cljc on classpath. Please check that namespaces with dashe
s use underscores in the Clojure file name.
...
Compilation failed: Subprocess failed
Uberjar aborting because jar failed: Compilation failed: Subprocess failed
Error encountered performing task 'uberjar' with profile(s): 'base,system,user,provided,dev,config'
Uberjar aborting because jar failed: Compilation failed: Subprocess failed
```

Unfortunately the same result:

```
C:\git\mranderson-test>lein do clean, inline-deps
project prefix:  mranderson43f30ef8
retrieve dependencies and munge clojure source files
in RESOLVED-TREE mode, working on a resolved dependency tree
 [clj-http "3.10.0"]
   [commons-io "2.6" :exclusions [[org.clojure/clojure]]]
   [org.apache.httpcomponents/httpmime "4.5.8" :exclusions [[org.clojure/clojure]]]
   [potemkin "0.4.5" :exclusions [[org.clojure/clojure]]]
     [clj-tuple "0.2.2"]
     [riddley "0.1.12"]
   [org.apache.httpcomponents/httpclient "4.5.8" :exclusions [[org.clojure/clojure]]]
     [commons-logging "1.2"]
   [org.apache.httpcomponents/httpcore "4.4.11" :exclusions [[org.clojure/clojure]]]
   [slingshot "0.12.2" :exclusions [[org.clojure/clojure]]]
   [org.apache.httpcomponents/httpclient-cache "4.5.8" :exclusions [[org.clojure/clojure]]]
   [org.apache.httpcomponents/httpasyncclient "4.1.4" :exclusions [[org.clojure/clojure]]]
     [org.apache.httpcomponents/httpcore-nio "4.4.10"]
   [commons-codec "1.12" :exclusions [[org.clojure/clojure]]]
unzipping [ clj-http  [ v3v10v0 ]]
unzipping [ httpmime  [ v4v5v8 ]]
unzipping [ commons-io  [ v2v6 ]]
unzipping [ potemkin  [ v0v4v5 ]]
unzipping [ clj-tuple  [ v0v2v2 ]]
unzipping [ httpclient-cache  [ v4v5v8 ]]
unzipping [ httpasyncclient  [ v4v1v4 ]]
unzipping [ httpclient  [ v4v5v8 ]]
unzipping [ commons-logging  [ v1v2 ]]
unzipping [ commons-codec  [ v1v12 ]]
unzipping [ httpcore-nio  [ v4v4v10 ]]
unzipping [ httpcore  [ v4v4v11 ]]
unzipping [ slingshot  [ v0v12v2 ]]
unzipping [ riddley  [ v0v1v12 ]]
  munge source files of riddley artifact on branch [] exposed false.
  munge source files of slingshot artifact on branch [] exposed false.
  munge source files of httpcore artifact on branch [] exposed false.
  munge source files of httpcore-nio artifact on branch [] exposed false.
  munge source files of commons-codec artifact on branch [] exposed false.
  munge source files of commons-logging artifact on branch [] exposed false.
  munge source files of httpclient artifact on branch [] exposed false.
  munge source files of httpasyncclient artifact on branch [] exposed false.
  munge source files of httpclient-cache artifact on branch [] exposed false.
  munge source files of clj-tuple artifact on branch [] exposed false.
    new ns: mranderson43f30ef8.clj-tuple.v0v2v2.clj-tuple
  munge source files of potemkin artifact on branch [] exposed false.
    new ns: mranderson43f30ef8.potemkin.v0v4v5.potemkin
  munge source files of commons-io artifact on branch [] exposed false.
  munge source files of httpmime artifact on branch [] exposed false.
  munge source files of clj-http artifact on branch [] exposed false.

C:\git\mranderson-test>lein do with-profile +plugin.mranderson/config uberjar
Compiling mranderson-test.core
Syntax error compiling at (core.clj:1:1).
Exception in thread "main" Syntax error compiling at (core.clj:1:1).
...
Caused by: java.io.FileNotFoundException: Could not locate clj_http/client__init.class, clj_http/client.clj or clj_http/client.cljc on classpath. Please check that namespaces with dashe
s use underscores in the Clojure file name.
...
Compilation failed: Subprocess failed
Uberjar aborting because jar failed: Compilation failed: Subprocess failed
Error encountered performing task 'uberjar' with profile(s): 'base,system,user,provided,dev,config'
Uberjar aborting because jar failed: Compilation failed: Subprocess failed
```

--- a/src/mranderson/core.clj
+++ b/src/mranderson/core.clj
@@ -213,7 +213,7 @@
                               (mapv fs/file))]
     (u/info (format "  munge source files of %s artifact on branch %s exposed %s." art-name-cleaned branch (boolean expose?)))
     (u/debug "    proj-source-dirs" project-source-dirs " clj files" clj-files "clj dirs" clj-dirs " path to dep" src-path "parent-clj-dirs: " parent-clj-dirs)
-    (u/debug "   modified namespace prefix: " repl-prefix)
+    (u/debug "    modified namespace prefix: " repl-prefix)
     (u/debug "    src path: " src-path)
     (u/debug "    parent clj dirs: " (str/join ":" parent-clj-dirs))
     (u/debug "    all dirs: " all-deps-dirs)

--- a/src/mranderson/core.clj
+++ b/src/mranderson/core.clj
@@ -232,9 +232,9 @@
             (move/replace-ns-symbol-in-source-files old-ns new-ns (u/file->extension (str clj-file)) project-source-dirs nil)))
         ;; a clj file without ns
         (when-not (= "project.clj" clj-file)
-          (let [old-path (str "target/srcdeps/" clj-file)
+          (let [old-path (fs/file srcdeps clj-file)
                 new-path (str (u/sym->file-name pprefix) "/" art-name-cleaned "/" art-version "/" clj-file)]
-            (fs/copy+ old-path (str "target/srcdeps/" new-path))
+            (fs/copy+ old-path (fs/file srcdeps new-path))
             ;; replace occurrences of file path references
             (doseq [file (u/clojure-source-files [srcdeps])]
               (update-path-in-file file clj-file new-path))

--- a/src/mranderson/core.clj
+++ b/src/mranderson/core.clj
@@ -241,13 +241,45 @@
             ;; remove old file
             (fs/delete old-path)))))))
 
+(defn- remove-invalid-duplicates!
+  "See issue #44. Some artifacts package duplicate files in weird locations
+   (e.g. riddley has `riddley.compiler` in the root of the package) which
+   causes a mismatch of expectations when moving files, resulting in
+   exceptions."
+  [srcdeps clj-files]
+  (let [by-ns (group-by
+                #(->> (fs/file srcdeps %)
+                      (read-file-ns-decl)
+                      (second))
+                clj-files)]
+    (mapcat
+      (fn [[nspace files]]
+        (if (and nspace (next files))
+          ;; A namespace has been duplicated, so we need to find the files that
+          ;; actually exist. There might be more than one file, in case of
+          ;; CLJ/CLJS duplication.
+          (let [expected-prefix (u/sym->file-name nspace)
+                match? #{(str expected-prefix ".clj")
+                         (str expected-prefix ".cljs")
+                         (str expected-prefix ".cljc")}]
+            (doseq [to-delete (remove match? files)]
+              (u/warn "  removing duplicated file with namespace mismatch:" to-delete)
+              (.delete (fs/file srcdeps to-delete)))
+            (filter match? files))
+          ;; Only one file means nothing to do (but it might still be in the
+          ;; wrong place).
+          files))
+      by-ns)))
+
 (defn- unzip-artifact! [{:keys [srcdeps]} {:keys [src-path branch]} dep]
   (let [art-name         (-> dep first name (str/split #"/") last)
         art-name-cleaned (str/replace art-name #"[\.-_]" "")
         art-version      (str "v" (-> dep second (str/replace "." "v")))
-        clj-files        (doall (unzip (-> dep meta :file) srcdeps))
+        _                (u/info "unzipping [" art-name-cleaned " [" art-version "]]")
+        clj-files        (->> (unzip (-> dep meta :file) srcdeps)
+                              (remove-invalid-duplicates! srcdeps)
+                              (doall))
         clj-dirs         (u/clj-files->dirs srcdeps clj-files)]
-    (u/info "unzipping [" art-name-cleaned " [" art-version "]]")
     (u/debug (format "resolving transitive dependencies for %s:" art-name))
     [{:art-name-cleaned art-name-cleaned
       :art-version      art-version

--- a/src/mranderson/move.clj
+++ b/src/mranderson/move.clj
@@ -242,7 +242,7 @@
   returns modified source as a string.
 
   Splits the source file, parses the ns macro if found to do all the necessary
-  transformations. Works on the body of namepsace as text as simpler transformations
+  transformations. Works on the body of namespace as text as simpler transformations
   are needed. When done puts the ns form and body back together."
   [content old-sym new-sym watermark extension-of-moved file-ext]
   (let [[ns-loc source-sans-ns] (split-ns-form-ns-body content)

--- a/src/mranderson/move.clj
+++ b/src/mranderson/move.clj
@@ -247,7 +247,8 @@
   [content old-sym new-sym watermark extension-of-moved file-ext]
   (let [[ns-loc source-sans-ns] (split-ns-form-ns-body content)
         opposite-platform       (util/platform-comp (util/extension->platform extension-of-moved))
-        [replaced-nodes ns-loc] (or (and (= ".cljc" file-ext)
+        [replaced-nodes ns-loc] (or (and ns-loc
+                                         (= ".cljc" file-ext)
                                          opposite-platform
                                          (find-and-replace-platform-specific-subforms opposite-platform ns-loc))
                                     [[] ns-loc])

--- a/test/mranderson/core_test.clj
+++ b/test/mranderson/core_test.clj
@@ -1,0 +1,67 @@
+(ns mranderson.core-test
+  (:require [mranderson.test :refer [with-mranderson]]
+            [clojure.java.io :as io]
+            [clojure.string :as string]
+            [clojure.test :refer :all]))
+
+;; ## Fixtures
+
+(def dependencies
+  '[^:inline-dep [org.clojure/data.xml "0.2.0-alpha6"]
+    ^:inline-dep [riddley "0.1.12"]
+    ^:inline-dep [cljfmt "0.7.0"]])
+
+(def clojure-namespace-file
+  {:path   "mrt/main.clj"
+   :content (str "(ns mrt.main\n"
+                 "  (:require [riddley.walk :as rw]"
+                 "            [clojure.data.xml :as xml]))")})
+
+(def clojure-resource-file
+  {:path   "mrt/resource.clj"
+   :content (pr-str [{:position {:x 1, :y 1}, :color :red}])})
+
+;; ## Helpers
+
+(defn- ->file
+  [{:keys [working-directory]} {:keys [path]}]
+  (io/file working-directory path))
+
+;; ## Tests
+
+(deftest t-resolved-tree-and-skip-java-repackage-classes
+  (with-mranderson
+    [project {:dependencies dependencies
+              :files        [clojure-namespace-file
+                             clojure-resource-file]
+              :opts         {:skip-repackage-java-classes true
+                             :unresolved-tree             false}}]
+    (let [{:keys [working-directory prefix ns-prefix]} project]
+
+      (testing "Clojure source file was correctly updated"
+        (let [riddley-prefix (str ns-prefix ".riddley.v0v1v12.riddley")
+              xml-prefix     (str ns-prefix ".dataxml.v0v2v0-alpha6.clojure.data.xml")
+              main-file      (->file project clojure-namespace-file)]
+          (is (= (-> (:content clojure-namespace-file)
+                     (string/replace "riddley" riddley-prefix)
+                     (string/replace "clojure.data.xml" xml-prefix))
+                 (slurp main-file)))))
+
+      (testing "Clojure resource file was not changed"
+        (let [resource-file (->file project clojure-resource-file)]
+          (is (= (:content clojure-resource-file)
+                 (slurp resource-file)))))
+
+      (testing "Dependency source file was correctly updated"
+        (let [riddley-file (io/file working-directory
+                                    prefix
+                                    "riddley"
+                                    "v0v1v12"
+                                    "riddley"
+                                    "walk.clj")]
+          (is (string/starts-with?
+                (slurp riddley-file)
+                (str "(ns ^{:mranderson/inlined true} " ns-prefix ".riddley.v0v1v12.riddley.walk\n"
+                     "  (:refer-clojure :exclude [macroexpand])\n"
+                     "  (:require\n"
+                     "    [" ns-prefix ".riddley.v0v1v12.riddley.compiler :as cmp]))"))))))))

--- a/test/mranderson/move_test.clj
+++ b/test/mranderson/move_test.clj
@@ -99,6 +99,10 @@
 
 (def ex-seven-cljs "(ns example.seven)")
 
+(def ex-data-readers
+  "{xml/ns clojure.data.xml.name/uri-symbol
+ xml/element clojure.data.xml.node/tagged-element}")
+
 (def medley-user-example
   "(ns example.user.medley
  (:require [medley.core :as medley]))")
@@ -159,6 +163,7 @@
         file-cljc     (create-source-file! (io/file example-dir "cross.cljc") ex-cljc)
         file-seven-clj  (create-source-file! (io/file example-dir "seven.clj") ex-seven-clj)
         file-seven-cljs (create-source-file! (io/file example-dir "seven.cljs") ex-seven-cljs)
+        file-data-readers (create-source-file! (io/file example-dir "data_readers.cljc") ex-data-readers)
         medley-dir   (io/file src-dir "medley")
         file-medley  (create-source-file! (io/file medley-dir "core.clj") medley-stub)
         file-medley-user (create-source-file! (io/file example-dir "user" "medley.clj") medley-user-example)
@@ -202,6 +207,8 @@
               "type of occurence is retained if keyword")
         (t/is (re-find #"\[example\.b\s*\[foo\]\s*\[bar\]\]" (slurp file-two))
               "prefixes should be replaced")
+        (t/is (= ex-data-readers (slurp file-data-readers))
+              "cljc file w/o ns macro is unchanged")
         (t/is (= ex-edn (slurp file-edn))
               "clj file wo/ ns macro is unchanged"))
 

--- a/test/mranderson/test.clj
+++ b/test/mranderson/test.clj
@@ -1,0 +1,98 @@
+(ns mranderson.test
+  (:require [mranderson.core :refer [mranderson]]
+            [me.raynes.fs :as fs]
+            [clojure.java.io :as io]
+            [clojure.test :as t])
+  (:import (java.io File)))
+
+;; ## Fixtures
+
+(def ^:private repositories
+  [["central" {:url "https://repo1.maven.org/maven2/", :snapshots false}]
+   ["clojars" {:url "https://repo.clojars.org/"}]])
+
+;; ## Helpers
+
+(defmacro ^:private with-temp-dir
+  "Create a temporary directory, run the body, then clean up the directory."
+  [sym & body]
+  `(let [f# (doto (File/createTempFile "mranderson" "")
+              (.delete)
+              (.mkdirs))
+         ~sym f#]
+     (try
+       (do ~@body)
+       (finally
+         (fs/delete-dir f#)))))
+
+(defn ^:private create-files!
+  "Create files and return root directories that contain them."
+  [working-directory files]
+  (->> (for [{:keys [path content]} files]
+         (let [dirname     (first (.split ^String path "/"))
+               target-file (io/file working-directory path)]
+           (.. target-file (getParentFile) (mkdirs))
+           (spit target-file  content)
+           (io/file working-directory dirname)))
+       (distinct)
+       (doall)))
+
+(defmacro ^:private with-project
+  "Set up a temporary directory with a structure that can be used by
+   mranderson."
+  [[sym files] & body]
+  `(with-temp-dir tmp#
+     (let [working-directory#  (doto (io/file tmp# "srcdeps")
+                                (.mkdirs))
+           source-directories# (create-files! working-directory# ~files)
+           prefix#             (str "mranderson_test" (rand-int 99999))
+           ~sym                {:prefix             prefix#
+                                :ns-prefix          (.replace prefix# "_" "-")
+                                :working-directory  working-directory#
+                                :source-directories source-directories#}]
+       ~@body)))
+
+(defn- project->context
+  "Convert the given project map into a context map for mranderson."
+  [{:keys [^File working-directory
+           source-directories
+           prefix]}
+   & [opts]]
+  (merge
+    {:pprefix                     prefix
+     :pname                       "mranderson-test"
+     :pversion                    "0.1.0-SNAPSHOT"
+
+     :project-source-dirs         source-directories
+     :srcdeps                     (.getPath working-directory)
+
+     :skip-repackage-java-classes false
+     :unresolved-tree             false
+
+     :watermark                   :mranderson/inlined
+     :expositions                 nil
+     :overrides                   nil
+     :prefix-exclusions           nil}
+    opts))
+
+(defn- project->paths
+  [{:keys [^File working-directory prefix]}]
+  "Convert the given project map into a paths map for mranderson."
+  {:src-path        (io/file working-directory prefix)
+   :parent-clj-dirs []
+   :branch          []})
+
+;; ## Run
+
+(defn with-mranderson*
+  [{:keys [dependencies files opts]} f]
+  (with-project [project files]
+    (mranderson repositories
+                dependencies
+                (project->context project opts)
+                (project->paths project))
+    (f project)))
+
+(defmacro with-mranderson
+  [[project {:keys [dependencies files opts] :as spec}] & body]
+  `(with-mranderson* ~spec (fn [~project] ~@body)))

